lib/AST/DiagnosticConsumer.cpp

```diff
@@ -196,19 +197,18 @@ bool FileSpecificDiagnosticConsumer::finishProcessing() {
   // behavior.
 
   bool hadError = false;
-  for (auto &subConsumer : SubConsumers)
-    hadError |= subConsumer.second && subConsumer.second->finishProcessing();
+  for (auto &subconsumer : Subconsumers)
+    hadError |= subconsumer.getConsumer() &&
+                subconsumer.getConsumer()->finishProcessing();
   return hadError;
 }
 
 void FileSpecificDiagnosticConsumer::
-    tellSubconsumersToInformDriverOfIncompleteBatchModeCompilation() const {
+    tellSubconsumersToInformDriverOfIncompleteBatchModeCompilation() {
   if (!HasAnErrorBeenConsumed)
     return;
-  for (auto &info : ConsumersOrderedByRange) {
-    if (!info.hasAnErrorBeenEmitted && info.consumer)
-      info.consumer->informDriverOfIncompleteBatchModeCompilation();
-  }
+  for (auto &info : ConsumersOrderedByRange)
```

Comment:
Maybe this should go in a separate commit, but this should also be iterating over Subconsumers instead of ConsumersOrderedByRange, so we get a stable iteration order between runs.

Reply:
Yes, I like it! Done!